ansible.mk

```diff
@@ -0,0 +1,192 @@
+
+# Code management.
+.PHONY: format tidy clean cli-doc lint
+
+format: ## Format the source code
+	$(Q)go fmt $(PKGS)
+
+tidy: ## Update dependencies
+	$(Q)go mod tidy -v
+
+clean: ## Clean up the build artifacts
+	$(Q)rm -rf build
+
+lint-dev:  ## Run golangci-lint with all checks enabled (development purpose only)
+	./hack/tests/check-lint.sh dev
+
+lint-fix: ## Run golangci-lint automatically fix (development purpose only)
+	./hack/tests/check-lint.sh fix
+
+lint: ## Run golangci-lint with all checks enabled in the ci
+	./hack/tests/check-lint.sh ci
+
+setup-k8s:
+	hack/ci/setup-k8s.sh $(K8S_VERSION)
+
+##############################
+# Generate Artifacts         #
+##############################
+
+##@ Generate
+
+.PHONY: generate gen-cli-doc gen-test-framework gen-changelog
+
+generate: gen-cli-doc  ## Run all non-release generate targets
+
+gen-cli-doc: ## Generate CLI documentation
+	./hack/generate/cli-doc/gen-cli-doc.sh
+
+gen-changelog: ## Generate CHANGELOG.md and migration guide updates
+	./hack/generate/changelog/gen-changelog.sh
+
+##############################
+# Build and Release                    #
+##############################
+
+##@ Build and Release
+
+# Build and release ansible-operator.
+.PHONY: release_builds release
+
+release_builds := \
+	build/ansible-operator-$(VERSION)-aarch64-linux-gnu \
+	build/ansible-operator-$(VERSION)-x86_64-linux-gnu \
+	build/ansible-operator-$(VERSION)-x86_64-apple-darwin \
+	build/ansible-operator-$(VERSION)-ppc64le-linux-gnu \
+	build/ansible-operator-$(VERSION)-s390x-linux-gnu
+
+# TODO: add `clean` recipe as the first step
+release: $(release_builds) $(release_builds:=.asc) ## Release ansible-operator
+
+build/ansible-operator-%-aarch64-linux-gnu: GOARGS = GOOS=linux GOARCH=arm64
+build/ansible-operator-%-x86_64-linux-gnu: GOARGS = GOOS=linux GOARCH=amd64
+build/ansible-operator-%-x86_64-apple-darwin: GOARGS = GOOS=darwin GOARCH=amd64
+build/ansible-operator-%-ppc64le-linux-gnu: GOARGS = GOOS=linux GOARCH=ppc64le
+build/ansible-operator-%-s390x-linux-gnu: GOARGS = GOOS=linux GOARCH=s390x
+build/ansible-operator-%-linux-gnu: GOARGS = GOOS=linux
+
+build/%: $(SOURCES) ## Build the ansible-operator binary
+	$(Q)$(GOARGS) go build $(GO_BUILD_ARGS) -o $@ ./cmd/ansible-operator
+
+build/%.asc: ## Create release signatures for ansible-operator release binaries
+	$(Q){ \
+	default_key=$$(gpgconf --list-options gpg | awk -F: '$$1 == "default-key" { gsub(/"/,""); print toupper($$10)}'); \
+	git_key=$$(git config --get user.signingkey | awk '{ print toupper($$0) }'); \
+	if [ "$${default_key}" = "$${git_key}" ]; then \
+		gpg --output $@ --detach-sig build/$*; \
+		gpg --verify $@ build/$*; \
+	else \
+		echo "git and/or gpg are not configured to have default signing key $${default_key}"; \
+		exit 1; \
+	fi; \
+	}
+
+##############################
+# Tests                      #
+##############################
+
+##@ Tests
+
+# Static tests.
+.PHONY: test test-sanity test-unit test-links
```

Comment:
We wiil not have the `test-links`. I understand that website will stick just in the SDK. 
The ansible/helm will use go docs only.

Reply:
Keeping ansible and helm docs in the SDK repo seems a little weird, although isn't that crazy an idea. @asmacdo @joelanford confirm?

Reply:
Ansible and Helm will not have CLI commands besides allow us to work with them as standalone mainly for development purpose. They are libs projects then, all can be doc via code and we will be use the godocs such as controller-runtime. When we update/bump its version in the SDK CLI then we can/must update its docs with the new features/changes and etc. I hope that it clarifies the idea.